src/main/java/ui/windows/PaneDelegator.java

```diff
@@ -0,0 +1,36 @@
+package ui.windows;
+
+import java.awt.event.ActionListener;
+import java.awt.event.ActionEvent;
+
+import ui.components.NavigationButton;
+
+public class PaneDelegator implements ActionListener {
+    private ICardLayoutManager manager;
+
+    /**
+     * Constructor for the PaneDelegator, which listens for an ActionEvent related to NavigationButtons
+     * and displays the window associated with the button click.
+     * @param manager The CardLayoutManager used to update the display the user sees.
+     */
+    public PaneDelegator(ICardLayoutManager manager) {
+        this.manager = manager;
+    }
+
+    /**
+     * If the user clicks a NavigationButton, the manager of the PaneDelegator will update the
+     * current window the user sees on screen.
+     * @param e The event to be processed.
+     */
+    @Override
+    public void actionPerformed(ActionEvent e) {
```

Comment:
Id add "throws IllegalArgumentException" to the method header.

Reply:
Good idea, let me add that now.